src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,82 @@
-export const Pagination = () => {};
+import classNames from 'classnames';
+import { FC } from 'react';
+
+type Props = {
+  total: number,
+  perPage: number,
+  currentPage: number,
+  onPageChange: (page: number) => void,
+};
+
+export const Pagination: FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const pageNums: number[] = [];
+  const pagesCounter = Math.ceil(total / perPage);
+
+  for (let i = 1; i <= pagesCounter; i += 1) {
+    pageNums.push(i);
+  }
+
+  const handlePageNumClick = (num: number) => {
+    if (num !== currentPage && num > 0 && num <= pageNums.length) {
+      onPageChange(num);
+    }
+  };
+
+  return (
+    <ul className="pagination">
+      <li className={classNames(
+        'page-item',
+        { disabled: currentPage === 1 },
+      )}
+      >
+        <a
+          data-cy="prevLink"
+          className="page-link"
+          href="#prev"
+          aria-disabled={currentPage === 1}
```

Comment:
you use the same expression 2 times, so better create a variable
```suggestion
    <ul className="pagination">
      <li className={classNames(
        'page-item',
        { disabled: someVariable },
      )}
      >
        <a
          data-cy="prevLink"
          className="page-link"
          href="#prev"
          aria-disabled={someVariable}
```